### PR TITLE
DAOS-10019 control: Move raft into its own package

### PR DESCRIPTION
**Pull request by mjmac**
Refactor the system package to remove the direct dependencies
on external raft/grpc packages in order to avoid bringing
them in for unrelated tools (e.g. daos, dmg, etc).

Signed-off-by: Michael MacDonald <mjmac.macdonald@intel.com>

### CODE REVIEW

File: src/control/system/member_test.go [hunk 1]
```diff
@@ -1,10 +1,10 @@
 //
-// (C) Copyright 2020-2021 Intel Corporation.
+// (C) Copyright 2020-2022 Intel Corporation.
 //
 // SPDX-License-Identifier: BSD-2-Clause-Patent
 //
 
-package system
+package system_test
```
Comment (kjacque) [inline src/control/system/member_test.go:7]: Specifically with the goal of testing from outside of the package?

**Review by kjacque — APPROVED** [review]
Good improvements.

File: src/control/system/member_test.go (hunk 1 above)
Comment (mjmac) [reply src/control/system/member_test.go:7]: Yes. We should be doing this as a general rule, IMO, unless we're testing unexported internal functions.

Comment (kjacque) [reply src/control/system/member_test.go:7]: Part of the reason we were testing from within the packages was that, and also to be able to twiddle with the unexported internals of some structs that might affect behavior. In some cases it doesn't make a difference, sure, but seems like we should establish a convention/strategy for when to use one or the other (and how to name the files in the case that we use both).

Comment (mjmac) [reply src/control/system/member_test.go:7]: Sure, I didn't mean to imply that it's wrong to do internal testing -- I just meant that I think we should probably do more testing from the outside to make us think a bit more about the exported surface of our packages.

**Review by tanabarr — APPROVED** [review]

**Pull request merged**